contracts/SynthetixRouterStrategy.sol

```diff
@@ -0,0 +1,351 @@
+            if (sUSDToWithdraw > 0) {
+                withdrawSomeWant(sUSDToWithdraw, true);
+            }
+        }
+    }
+
+    function liquidatePosition(uint256 _amountNeeded)
+        internal
+        override
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        uint256 wantBal = balanceOfWant(); // want is always sUSD
+
+        if (wantBal < _amountNeeded) {
+            (_liquidatedAmount, _loss) = withdrawSomeWant(
+                _amountNeeded.sub(wantBal),
+                false
+            );
+        }
+
+        _liquidatedAmount = Math.min(
+            _amountNeeded,
+            _liquidatedAmount.add(wantBal)
+        );
+    }
+
+    function updateSUSDBuffer(uint256 _susdBuffer) public onlyVaultManagers {
+        require(_susdBuffer <= 10_000, "!too high");
+        susdBuffer = _susdBuffer;
+    }
+
+    function checkWaitingPeriod() private returns (bool freeToMove) {
+        return
+            _exchanger().maxSecsLeftInWaitingPeriod(
+                address(this),
+                synthCurrencyKey
+            ) == 0;
+    }
+
+    function depositInVault() public onlyVaultManagers {
+        uint256 balanceOfSynth = _balanceOfSynth();
+        if (balanceOfSynth > DUST_THRESHOLD && checkWaitingPeriod()) {
+            _checkAllowance(
+                address(yVault),
+                address(_synthCoin()),
+                balanceOfSynth
+            );
+            yVault.deposit();
+        }
+    }
+
+    //safe to enter more than we have
+    function withdrawSomeWant(uint256 _amount, bool performExchanges)
+        private
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        if (performExchanges) {
+            // we exchange synths to susd
+            uint256 synthBalanceBefore = _balanceOfSynth();
+            uint256 sUSDBalanceBefore = balanceOfWant();
+            uint256 _new_amount = _amount.sub(sUSDBalanceBefore);
+
+            uint256 _synth_amount = _synthFromSUSD(_new_amount);
+            uint256 _remaining_amount = _synth_amount;
+            if (checkWaitingPeriod()) {
+                if (_synth_amount <= synthBalanceBefore) {
+                    exchangeSynthToSUSD(_synth_amount);
+                    return (_amount, 0);
+                }
+
+                _remaining_amount = _synth_amount.sub(synthBalanceBefore);
+            }
+            _withdrawFromYVault(_remaining_amount);
+            uint256 newBalanceOfSynth = _balanceOfSynth();
+            if (newBalanceOfSynth > DUST_THRESHOLD) {
+                exchangeSynthToSUSD(newBalanceOfSynth);
+            }
+        }
+
+        uint256 totalAssets = balanceOfWant();
+        if (_amount > totalAssets) {
+            _liquidatedAmount = totalAssets;
+            _loss = _amount.sub(totalAssets);
+        } else {
+            _liquidatedAmount = _amount;
+        }
+    }
+
+    function liquidateAllPositions()
```

Comment:
Yes, we talked about this with @poolpitako and the idea was to revert in case you have not freed all the investments before. ideally, manualremovefullLiquidity needs to be called before and the strategy should have everything in susd

Reply:
gotcha